addons/payment/views/payment_templates.xml

```diff
@@ -56,14 +56,79 @@
             <t t-set="footer_template_id"
                t-value="footer_template_id or 'payment.footer'"/>
 
+            <!-- === Tokens === -->
+            <t t-if="token_count>0">
+                <div class="mt-3 fs-4">Saved payment methods</div>
+                <div class="card">
+                    <t t-foreach="tokens" t-as="token">
+                        <t t-set="provider" t-value="token.provider_id"/>
```

Comment:
It's a good idea to extract `token.provider_id` in a variable, but you should use it more. There are several occurrences of `token.provider_id` that still coexist with `provider`, where I'd expect to find only `provider`.